pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -279,11 +293,43 @@ func (a *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) error
 			Reason:  remoteFinishedCond.Reason,
 			Message: fmt.Sprintf("From remote %q: %s", remote, remoteFinishedCond.Message),
 		})
-		return a.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
+		return reconcile.Result{}, a.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
 	}
 
 	hasReserving, reservingRemote := group.FirstReserving()
 
+	// if the reserving workload is out of sync, remove it
+	if hasReserving {
+		outOfSync := group.local == nil || !equality.Semantic.DeepEqual(group.local.Spec, group.remotes[reservingRemote].Spec)
+		if outOfSync {
+			if err := client.IgnoreNotFound(group.RemoveRemoteObjects(ctx, reservingRemote)); err != nil {
+				log.V(2).Error(err, "Deleting out of sync remote objects", "remote", reservingRemote)
+				return reconcile.Result{}, err
+			}
+			group.remotes[reservingRemote] = nil
+
+			// check if another one exists
+			hasReserving, reservingRemote = group.FirstReserving()
+		}
+	}
+
+	// If there is no reserving and the AC is ready, the connection with the reserving remote might
+	// be lost, keep the workload admitted for keepReadyTimeout and put it back in the queue after that.
+	if !hasReserving && acs.State == kueue.CheckStateReady {
```

Comment:
Why do we differ the logic based on !hasReserving?
I think the main use case for the timeout is to prevent requeue of a running workload (admitted locally and remotely), in case of transient connectivity loss. I thought in case of such workloads we have hasReserving=true.